llama-index-core/llama_index/core/workflow/context_serializers.py

```diff
@@ -0,0 +1,81 @@
+import base64
+import json
+import pickle
+from abc import ABC, abstractmethod
+from typing import Any
+from pydantic import BaseModel
+
+from llama_index.core.schema import BaseComponent
+from .utils import import_module_from_qualified_name, get_qualified_name
+
+
+class BaseSerializer(ABC):
+    @abstractmethod
+    def serialize(self, value: Any) -> str:
+        ...
+
+    @abstractmethod
+    def deserialize(self, value: str) -> Any:
+        ...
+
+
+class JsonSerializer(BaseSerializer):
+    def serialize(self, value: Any) -> str:
+        if isinstance(value, BaseComponent):
+            return json.dumps(
+                {
+                    "__is_component": True,
+                    "value": value.to_dict(),
+                    "qualified_name": get_qualified_name(value),
+                }
+            )
+        elif isinstance(value, BaseModel):
+            return json.dumps(
+                {
+                    "__is_pydantic": True,
+                    "value": value.model_dump(),
+                    "qualified_name": get_qualified_name(value),
+                }
+            )
+
+        return json.dumps(value)
+
+    def deserialize(self, value: str) -> Any:
+        try:
+            data = json.loads(value)
+
+            if (
+                isinstance(data, dict)
+                and data.get("__is_pydantic")
+                and data.get("qualified_name")
+            ):
+                module_class = import_module_from_qualified_name(data["qualified_name"])
+                return module_class.model_validate(data["value"])
+            elif (
+                isinstance(data, dict)
+                and data.get("__is_component")
+                and data.get("qualified_name")
+            ):
+                module_class = import_module_from_qualified_name(data["qualified_name"])
+                return module_class.from_dict(data["value"])
+        except Exception as e:
+            breakpoint()
```

Comment:
do we need `breakpoint` here. Do we want to raise?

Reply:
oh lol leftover debugging, whoops